packages/compiler/src/server/serverlib.ts

```diff
@@ -347,6 +367,91 @@ export function createServer(host: ServerHost): Server {
     }
   }
 
+  async function renameFiles(params: RenameFilesParams): Promise<void> {
+    const firstFilePath = params.files[0];
+    if (!firstFilePath) {
+      return;
+    }
+
+    // Update cache for renamed folders.
+    for (const file of params.files) {
+      const oldFilePath = await fileService.getPath({ uri: file.oldUri });
+      const newFilePath = await fileService.getPath({ uri: file.newUri });
+      const isDirRename = !file.oldUri.endsWith(".tsp");
+
+      if (isDirRename) {
+        const files = await listAllFilesInDir(compilerHost, newFilePath);
+        for (const file of files) {
+          const oldFile = resolvePath(oldFilePath, file);
+
+          fileSystemCache.notify([
+            { uri: fileService.getURL(oldFile), type: FileChangeType.Deleted },
+          ]);
+        }
+      }
+    }
+
+    const mainFile = await compileService.getMainFileForDocument(
+      await fileService.getPath({ uri: firstFilePath.newUri }),
+    );
+
+    // Add this method to resolve timing issues between renamed files and `fs.stat`
+    // to prevent `fs.stat` from getting the files before modification.
+    // Currently the test requires a delay of 300ms
+    await new Promise((resolve) => setTimeout(resolve, 300));
```

Comment:
want to try one more thing here so we don't have to have this delay which slows down test

Reply:
meh, wanted to see if we could provide aliases when resolving files in the host during that compile so it treats the before and after paths as the same and ignore timing issue but that would be quite a bigger change I think